File: diploma-repository/src/main/java/org/arhor/diploma/repository/domain/Account.java
package org.arhor.diploma.repository.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "Account")
@EqualsAndHashCode(callSuper = true)
public class Account extends AuditableDomainObject<Long> {

  @Column(unique = true, nullable = false, length = 30)
  private String username;

  @Column(nullable = false, length = 512)
  private String password;

  @Column(unique = true, length = 128)
  private String email;

  @Column(nullable = false, length = 60)
  private String firstName;

  @Column(nullable = false, length = 60)
  private String lastName;

  @Column(length = 10)
  private String role;
}

File: diploma-server/server-data/src/main/kotlin/org/arhor/diploma/data/persistence/domain/Account.kt
package org.arhor.diploma.data.persistence.domain

import org.arhor.diploma.data.classBasedStaticHashCode
import org.arhor.diploma.data.persistence.domain.core.AuditableDomainObject
import org.arhor.diploma.data.persistence.domain.core.DomainObject
import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import javax.persistence.*

@Entity
@Table(name = "accounts")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
data class Account(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "${DomainObject.SEQ_GEN_NAME}_accounts")
    @SequenceGenerator(
        name = "${DomainObject.SEQ_GEN_NAME}_accounts",
        sequenceName = "accounts_id_seq",
        allocationSize = DomainObject.SEQ_ALLOC_SIZE
    )
    @Column(name = "id", nullable = false, updatable = false)
    override var id: Long? = null,

    @Column(unique = true, nullable = false, length = 30)
    var username: String? = null,

    @Column(nullable = false, length = 512)
    var password: String? = null,

    @Column(unique = true, length = 128)
    var email: String? = null,

    @Column(nullable = false, length = 60)
    var firstName: String? = null,

    @Column(nullable = false, length = 60)
    var lastName: String? = null,
) : AuditableDomainObject<Long>() {

    override val tableName: String
        get() = "accounts"

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "profile_id")
    var securityProfile: SecurityProfile? = null

    override fun hashCode(): Int {
        return classBasedStaticHashCode()
    }
}
